src/reweighting.hxx

```diff
@@ -92,4 +93,54 @@ auto topptreweighting(auto &df, const std::string &weightname,
                          {gen_pdgids, gen_status, gen_pt});
     return df1;
 }
+
+/**
+ * @brief Function used to evaluate Z pt mass weights
+ *
+ * @param df The input dataframe
+ * @param weightname name of the generated weight
+ * @param gen_boson name of the column that contains a pair of Lorentzvectors,
+ * where the first one is the one of the genboson
+ * @param workspace_file path to the file which contains the workspace to be
+ * read
+ * @param functor_name name of the function from the workspace
+ * @param argset arguments of the function
+ * @return a new dataframe containing the new column
+ */
+auto zPtMassReweighting(auto &df, const std::string &weightname,
+                        const std::string &gen_boson,
+                        const std::string &workspace_file,
+                        const std::string &functor_name,
+                        const std::string &argset) {
+
+    // retrieve pt and mass of gen boson reconstructed with the method used by
+    // recoil corrections; resulting quantities are only for the purpose of this
+    // method
+    auto df1 = df.Define(gen_boson + "_pt",
+                         [](const std::pair<ROOT::Math::PtEtaPhiMVector,
+                                            ROOT::Math::PtEtaPhiMVector> &p4) {
+                             return (float)p4.first.pt();
+                         },
+                         {gen_boson});
+    auto df2 = df1.Define(gen_boson + "_mass",
```

Comment:
maybe add `internal` `zpt_` or something to the name, so it wont block the `genboson_pt` column for the version where the genboson is read out directly